src/typeahead/__tests__/Typeahead.test.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react'
+import { mount } from 'enzyme'
+import transformOptions from '../transformOptions'
+
+import Typeahead from '../Typeahead'
+
+const selectOptions = [
+  { value: '1234', label: 'Chocolate', subLabel: 'mint' },
+  { value: '5678', label: 'Strawberry', subLabel: 'Cream' },
+  { value: '9876', label: 'Vanilla', subLabel: 'Vanilla' },
+]
+
+const WrappedSelectTypeahead = () => {
+  const [inputValue] = useState('Choc')
+
+  return (
+    <Typeahead
+      menuIsOpen={true}
+      inputValue={inputValue}
+      isMulti={false}
+      closeMenuOnSelect={false}
+      label="Standard options"
+      name="test_1"
+      options={transformOptions(selectOptions)}
+      placeholder="Search..."
+    />
+  )
+}
+
+describe('Typeahead', () => {
+  let wrapper
+  describe('<Select>', () => {
+    describe('when searching with the characters "choc"', () => {
+      wrapper = mount(<WrappedSelectTypeahead />)
+      const highlightedTextArray = wrapper.find('[data-test-id="test-1"]')
+      const highlightedText = highlightedTextArray.map((x) => x.text())[0]
+      test('should highlight the matching string', () => {
+        expect(highlightedText).toEqual('Choc')
+      })
+    })
+  })
+})
```

Comment:
We can add a bit more tests, here's a bare minimum:

```suggestion
import React from 'react'
import { mount } from 'enzyme'
import transformOptions from '../transformOptions'

import Typeahead from '../Typeahead'
import Highlighter from '../Highlighter'

const OPTIONS = [
  { value: '1234', label: 'Chocolate', subLabel: 'mint' },
  { value: '5678', label: 'Strawberry', subLabel: 'Cream' },
  { value: '9876', label: 'Vanilla', subLabel: 'Vanilla' },
]

const BASIC_PROPS = {
  name: 'test',
  options: transformOptions(OPTIONS),
  menuIsOpen: true,
}

describe('Typeahead', () => {
  test('highlights typed text', () => {
    const selectWrapper = mount(
      <Typeahead {...BASIC_PROPS} inputValue="choc" />
    )

    expect(selectWrapper.find(Highlighter)).toHaveLength(3)

    expect(selectWrapper.text()).toEqual(
      'Select...Chocolate - mintStrawberry - CreamVanilla - Vanilla'
    )

    expect(
      selectWrapper
        .find(Highlighter)
        .at(0)
        .find('span')
        .text()
    ).toEqual('Choc')
  })
})

```

Reply:
Thanks for this, I have replaced your tests over mine but I can't think of how I can do more tests. Ideally we need it to simulate the typing of text in the input but it just won't play nicely. The above is much cleaner so I used it but it doesn't give me much more than what I had apart from asserting 3 options which we already know we have. It would be nice to see that when we search for "Choc" we have one option left and its characters "Choc" are highlighted. I would also like to test the async version but I can't type in the text field to fire the onChange handler :-(

Reply:
Looks like `react-select` doesn't test well https://stackoverflow.com/questions/41991077/testing-react-select-component

Reply:
Although, you could add tests for the custom props, e.g. `reduceSize`

Reply:
After talking with Peter we have removed the label and hint out of this component as it made sense not to bake this in so these props are removed now.